variables.tf

```diff
@@ -0,0 +1,75 @@
+variable "aggregation_dimensions" {
+  description = "Specifies the dimensions that collected metrics are to be aggregated on."
+  type        = "list"
+
+  default = [
+    ["InstanceId"],
+    ["AutoScalingGroupName"],
+  ]
+}
+
+variable "cpu_resources" {
+  description = "Specifies that per-cpu metrics are to be collected. The only allowed value is *. If you include this field and value, per-cpu metrics are collected"
+  type        = "string"
+  default     = "\"resources\": [\"*\"],"
+}
+
+variable "disk_resources" {
+  description = "If you specify an array of devices, CloudWatch collects metrics from only those devices. Otherwise, metrics for all devices are collected"
+  type        = "list"
+  default     = ["/"]
+}
+
+variable "userdata_part_content" {
+  description = "The user data that should be passed along from the caller of the module."
+  type        = "string"
+  default     = ""
+}
+
+variable "userdata_part_content_type" {
+  description = "What format is userdata_part_content in - eg 'text/cloud-config' or 'text/x-shellscript'."
+  type        = "string"
+  default     = "text/cloud-config"
+}
+
+variable "userdata_part_merge_type" {
+  description = "Control how cloud-init merges user-data sections."
+  type        = "string"
+  default     = "list(append)+dict(recurse_array)+str()"
+}
+
+variable "namespace" {
+  description = "Namespace, which could be your organization name or abbreviation, e.g. 'eg' or 'cp'"
+  type        = "string"
+  default     = ""
+}
+
+variable "environment" {
+  description = "Environment, e.g. 'prod', 'staging', 'dev', 'pre-prod', 'UAT'"
+  type        = "string"
+  default     = ""
```

Comment:
remove the default and make it required ? (to be consistent with other CP modules)